src/main/java/defusion/EuropeanUnion.java

```diff
@@ -0,0 +1,95 @@
+package defusion;
+
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class EuropeanUnion {
+
+    @Getter
+    private List<Country> countries;
+    private Map<Integer, City> cities = new HashMap<>();
+
+    private EuropeanUnion(List<Country> countries) {
+        this.countries = countries;
+        this.countries.forEach(this::createCities);
+
+        for (City city : cities.values()) {
+            int x = city.getPosition().getX();
+            int y = city.getPosition().getY();
+
+            city.addNeighbourIfPresent(cities.get(City.positionHash(x - 1, y)));
+            city.addNeighbourIfPresent(cities.get(City.positionHash(x + 1, y)));
+            city.addNeighbourIfPresent(cities.get(City.positionHash(x, y + 1)));
+            city.addNeighbourIfPresent(cities.get(City.positionHash(x, y - 1)));
+
+            city.createAccounts(countries);
+        }
+
+        if(countries.size() > 1 && countries.stream().anyMatch(Country::isIsolated)) {
```

Comment:
Maybe better way is to add exception message to the output file?

Reply:
ready